packages/vm/src/cache.rs

```diff
@@ -47,20 +56,22 @@ where
     /// This function is marked unsafe due to `FileSystemCache::new`, which implicitly
     /// assumes the disk contents are correct, and there's no way to ensure the artifacts
     //  stored in the cache haven't been corrupted or tampered with.
-    pub unsafe fn new<P: Into<PathBuf>>(
-        base_dir: P,
-        supported_features: HashSet<String>,
-    ) -> VmResult<Self> {
-        let base = base_dir.into();
-        let wasm_path = base.join(WASM_DIR);
+    pub unsafe fn new(options: CacheOptions) -> VmResult<Self> {
+        let CacheOptions {
```

Comment:
Nice. I was wondering how to do such unstructuring and it failed without the type on the left. I will try this sometime.